docs/DeveloperGuide.adoc

```diff
@@ -337,30 +373,123 @@ Use case ends.
 * 3a. The given index is invalid.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1. Scheduler shows an error message.
 +
 Use case resumes at step 2.
 
-_{More to be added}_
+[discrete]
+=== Use case: (UC-2) Import interviewee's time slots
+
+*MSS*
+
+1.  User requests to import interviewee's time slots and specify file location
+2.  Scheduler imports interviewee's time slots from specified file
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. Specified file doesn't exist
+** 1a1. Scheduler shows error message
++
+Use case resumes at step 1.
+
+* 1b. Specified file doesn't follow correct format
+** 1b1. Scheduler shows error message
++
+Use case resumes at step 1.
+
+
+[discrete]
+=== Use case: (UC-3) Import interviewer's time slots
+
+*MSS*
+
+1.  User requests to import interviewer's time slots and specify file location
+2.  Scheduler imports interviewer's time slots from specified file
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. Specified file doesn't exist
+** 1a1. Scheduler shows error message
++
+Use case resumes at step 1.
+
+* 1b. Specified file doesn't follow correct format
+** 1b1. Scheduler shows error message
++
+Use case resumes at step 1.
+
+
+[discrete]
+=== Use case: (UC-4) Schedule interview slots
+
+
+*MSS*
+
+1. User __imports interviewee's availability (UC-2)__
+2. User __imports interviewer's availability (UC-3)__
+3.  User requests to generate timetable of all available interview time slots based on the availability of *interviewers*.
+4.  Scheduler generates timetable
+5.  User requests schedule interviewees based on the available timetable
+6.  Scheduler allocates interviewees into the time slots in the generated timetable
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 5a. Unable to fit all interviewees into timetable.
+** 5a1. Scheduler shows an error message with the names of interviewees that are not allocated a slot.
++
+Use case ends.
 
 [appendix]
 == Non Functional Requirements
 
-.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  The software should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
+.  Should be able to hold up to 1000 interviewers and interviewees without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-_{More to be added}_
+.  The software should be simple enough to use without much manual intervention by the user.
```

Comment:
What do you mean by manual intervention? Isn't entering commands manual intervention?